File: FirstTask/JavaScript/newton_sqrt/newtonsqrtAlgo.js
export function newton_sqrt(number){
    let first = number;
    let latest = number;
    i = 1;
    while (i < 1000 && (latest * latest - first) > 0.0001) {
        latest = 0.5 * (latest + first / latest);
        i++;
    }
    return latest;
}

File: FirstTask/C++/newton_sqrt/newton_sqrtAlgo.cpp
#include "newton_sqrtAlgo.h"

#include <iostream>

double newton_sqrt(double number, double first) {
    if (number < 0 || first < 0) {
        std::cerr << "Error in input: Negative numbers not possible" << std::endl;
    }

    return (0.5 * (number + first / number));
}
